compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/Transforms.cpp

```diff
@@ -354,6 +354,30 @@ collapsibleSlicePrecondition(RewriterBase &rewriter,
   return success();
 }
 
+static Operation *
+setInsertionPointToLastIndexOperand(RewriterBase &rewriter,
+                                    tensor::ParallelInsertSliceOp op) {
+  DominanceInfo domInfo;
+  auto subsetOp = cast<SubsetInsertionOpInterface>(op.getOperation());
+  SmallVector<Value> values = subsetOp.getValuesNeededToBuildSubsetExtraction();
+  Operation *lastOp = nullptr;
+  for (auto val : values) {
+    auto definingOp = val.getDefiningOp();
+    if (!definingOp) {
+      definingOp = cast<BlockArgument>(val).getParentBlock()->getParentOp();
+    }
+    if (!lastOp || domInfo.dominates(lastOp, definingOp)) {
```

Comment:
Silly question: wouldn't it make sense to find the insertion point and then to only move once?